src/http/chain-registry/ChainRegistry.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Class that enables interaction with the Contract Registry
+ */
+
+import { HttpClient } from '../client/http'
+import { IChain, IHttpGETRequest, IHttpPOSTRequest, IHttpResponse, IRegisterChainRequest } from '../types'
+
+export class ChainRegistry {
+  private client: HttpClient
+
+  /**
+   * Creates a new ChainRegistry instance
+   *
+   * @param chainRegistryHost - the URL and port of the chain registry RPC
+   */
+  constructor(chainRegistryHost: string) {
+    this.client = new HttpClient({
+      host: chainRegistryHost
+    })
+  }
+
+  public async chains(authToken?: string): Promise<IChain[]> {
+    const req: IHttpGETRequest = {
+      path: '/chains',
+      authToken
+    }
+
+    const res: IHttpResponse = await this.client.get(req, {})
```

Comment:
this really seems "go like" code. Why isn't the get function just rejecting the promise ?

Reply:
As I decided to implement, it is a bit `go:like`, but it is a pattern itself where you let users to handle expected errors as in this case it would be a !== 200 response. But I am open to discuss another alternatives